deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/Xception.java

```diff
@@ -0,0 +1,238 @@
+package org.deeplearning4j.zoo.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
+import org.deeplearning4j.nn.api.Model;
+import org.deeplearning4j.nn.api.OptimizationAlgorithm;
+import org.deeplearning4j.nn.conf.*;
+import org.deeplearning4j.nn.conf.distribution.NormalDistribution;
+import org.deeplearning4j.nn.conf.graph.ElementWiseVertex;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.conf.layers.*;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.deeplearning4j.zoo.ModelMetaData;
+import org.deeplearning4j.zoo.PretrainedType;
+import org.deeplearning4j.zoo.ZooModel;
+import org.deeplearning4j.zoo.ZooType;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.learning.config.AdaGrad;
+import org.nd4j.linalg.learning.config.IUpdater;
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+
+/**
+ * U-Net
+ *
+ * An implementation of Xception in Deeplearning4j. A novel deep convolutional neural network architecture inspired by Inception, where Inception modules have been replaced with depthwise separable convolutions.
+ *
+ * <p>Paper: https://arxiv.org/abs/1610.02357</p>
+ * <p>ImageNet weights for this model are available and have been converted from https://keras.io/applications/.</p>
+ *
+ * @author Justin Long (crockpotveggies)
+ *
+ */
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Xception extends ZooModel {
+
+    @Builder.Default private long seed = 1234;
+    @Builder.Default private int[] inputShape = new int[] {3, 299, 299};
+    private int numClasses;
+    @Builder.Default private WeightInit weightInit = WeightInit.RELU;
+    @Builder.Default private IUpdater updater = new AdaGrad(1e-4);
```

Comment:
Is this correct? I'm not familiar with the architecture, but AdaGrad is a curious choice for the default...
(Looks like they use momentum and RMSProp in the paper with a LR decay)